src/server.cpp

```diff
@@ -1644,14 +1644,21 @@ void Server::SendChatMessage(u16 peer_id, const std::wstring &message)
 {
 	DSTACK(FUNCTION_NAME);
 
+	std::wstring processed_message;
+	if (m_clients.getProtocolVersion(peer_id) < 27)
+		processed_message = unescape_enriched(message);
+	else
+		processed_message = message;
+
 	NetworkPacket pkt(TOCLIENT_CHAT_MESSAGE, 0, peer_id);
-	pkt << message;
+	pkt << processed_message;
 
 	if (peer_id != PEER_ID_INEXISTENT) {
 		Send(&pkt);
 	}
 	else {
-		m_clients.sendToAll(&pkt);
+		for (u16 id: m_clients.getClientIDs())
```

Comment:
please change function to be properly recursive if needed, here we create a packet and verify the peer_id in broadcast mode, please move packet ceration into unicast mode and the proto version check